Add documentation for feature in #92

Rudd-O opened the pull request:
https://github.com/plone/plone-backend/pull/92/files related.

Rudd-O committed:
Add documentation for feature in #92

https://github.com/plone/plone-backend/pull/92/files related.

Rudd-O committed:
Document the package-includes feature too.

Had this for a while, never actually saw documentation for it.

stevepiercy commented on docs/install/containers/images/backend.md:
```diff
@@ -65,6 +65,13 @@ Plone 6 example:
+With the standard container, it is not possible to add configuration fragments to
+`zope.conf` directly or add extra ZCML, like it is with the `buildout` deployment
+method.  However, you can derive your own container image, and drop in configuration
+fragments.  See *Extending from this image* below for instructions.
```
Change this to a `ref`.

```suggestion
fragments.  See {ref}`backend-extending-from-this-image-label` below for instructions.
```

Also above the section heading below, insert a label.

```
(backend-extending-from-this-image-label)=

## Extending from this image
```

Rudd-O replied:
It's done.

stevepiercy replied:
I don't see the label inserted on line 288. Did you push the commit?

Rudd-O replied:
I merged it directly via Github.  WTF.

stevepiercy replied:
No, that line you had do separately. I was not able to insert a suggestion in something you had not edited. /me shakes fist at GitHub UI not doing everything under the sun.

stevepiercy commented on docs/install/containers/images/backend.md:
```diff
@@ -304,6 +311,34 @@ And start a container.
+In the directory containing your `Dockerfile`, create a folder `etc/zope.conf.d`.
+Add your `zope.conf` configuration fragments there.
+
+Now add the following to your `Dockerfile`:
```
Where should it be added? Without guidance, some people will place it in an unfortunate point in the Dockerfile and not take advantage of skipping over steps that have not changed.

Rudd-O replied:
Good point.  Will fix.

stevepiercy commented on docs/install/containers/images/backend.md:
````diff
@@ -304,6 +311,34 @@ And start a container.
+Now add the following to your `Dockerfile`:
+
+```Dockerfile
+COPY /etc/package-includes/*.zcml /app/etc/package-includes/
````
Where should this be placed in the Dockerfile?

Rudd-O replied:
Good point.  Will fix.

stevepiercy reviewed (changes requested):
Minor enhancement, and questions about where to place commands inside Dockerfile.

Rudd-O committed:
Update docs/install/containers/images/backend.md

Co-authored-by: Steve Piercy <web@stevepiercy.com>

Rudd-O committed:
Where must we add?

Rudd-O committed:
Fix label and reorient where the comments are.

Rudd-O commented:
The ref should be there now @stevepiercy .

Rudd-O committed:
Reorg of the add-on variables section and minor other changes for consistency.

stevepiercy reviewed (dismissed):
There we go!

stevepiercy commented:
I'm not sure why the test check failed. I restarted it. Fingers crossed.

stevepiercy commented:
Passing now. Will merge when https://github.com/plone/plone-backend/pull/92 is merged. Please ping me when ready.

Rudd-O commented:
@stevepiercy I am ready.  #92 in plone-backend is not yet approved tho.

Rudd-O committed:
Turns out RUN needs to run before the COPY.

The `apt-get` upgrade recommendation per the documentation should always run before adding any other code.

Rudd-O committed:
Add documentation on overriding variables in derived containers.

Strengthening the case for deriving containers from the base image.

Rudd-O commented:
It says "merge blocked. @stevepiercy 

![image](https://user-images.githubusercontent.com/346306/201933866-6b875f6e-2226-4bc9-a803-30871c4e6bba.png)

stevepiercy commented:
@Rudd-O correct. Only members of @plone/documentation-team and Organization administrators, repository administrators, and users with the Maintain role, have permission to merge. Once https://github.com/plone/plone-backend/pull/92 is merged, then ping me, and I will merge this one. Or are they waiting for me to merge this one before they merge that one?

jensens committed:
Merge branch '6-dev' into zopeconfd
